metrics/prometheus.go

```diff
@@ -0,0 +1,155 @@
+package metrics
+
+import (
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "traefik"
+)
+
+// TODO: #156 - More detailed metrics
+type entrypointMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type providerMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type frontendMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type backendMetrics struct {
+}
+
+type globalMetrics struct {
+	UpTime                    prometheus.Gauge
+	RequestResponseTimeTotal  prometheus.Gauge
```

Comment:
I don't think so. The value is already a sum, and is extracted from [github.com/thoas/stats](http://github.com/thoas/stats) with a ```set()```, not an ```inc()```.

In a futur step, we will increment this metric from a middleware at each single request. So using a counter will make sense.

We will also use labels to isolate metrics by backends, frontends, services, entrypoints... Then Prometheus and Grafana will be able to deduce the total response time and the difference between two scrapes of the same counter.

WDYT ?